## Trades
In this notebook we will look further into the trade data scraped from the CapFriendly website.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

After our initial imports, we will first download the csv file containing the scraped data.

In [51]:
raw = pd.read_csv('/Users/phil/Documents/Capstone/Data/test_df_all.cvs')
raw

Unnamed: 0                 0  \
0              0              DATE   
1              1              DATE   
2              2     Jun. 30, 2006   
3              3     Jun. 26, 2006   
4              4     Jun. 24, 2006   
...          ...               ...   
1805         151      Jul. 7, 2022   
1806         152      Jul. 7, 2022   
1807         153      Jul. 7, 2022   
1808         154      Jul. 3, 2022   
1809         155  No Results Found   

                                                      1   2  \
0                                         TRADE DETAILS NaN   
1                                         TRADE DETAILS NaN   
2     Columbus Blue Jackets Acquire:Freddy ModinFred... NaN   
3     Arizona Coyotes Acquire:Nick Boynton2007 4th r... NaN   
4                     Boston Bruins Acquire:Tuukka Rask NaN   
...                                                 ...  ..   
1805  Montreal Canadiens Acquire:2022 1st round pick... NaN   
1806  Chicago Blackhawks Acquire:2022 1st round pick... NaN   
1807  Colorado Avalanche Acquire:Alexandar Georgiev ... NaN   
1808  Nashville Predators Acquire:Ryan McDonagh · $6... NaN   
1809                                                NaN NaN   

                                                      3   4    5  
0                                                   NaN NaN  NaN  
1                                                   NaN NaN  NaN  
2                Tampa Bay Lightning Acquire:Marc Denis NaN  NaN  
3     Boston Bruins Acquire:Paul Mara2007 3rd round ... NaN  NaN  
4           Toronto Maple Leafs Acquire:Andrew Raycroft NaN  NaN  
...                                                 ...  ..  ...  
1805  New York Islanders Acquire:Alexander Romanov ·... NaN  NaN  
1806  Ottawa Senators Acquire:Alex DeBrincat · $6,40... NaN  NaN  
1807  New York Rangers Acquire:2022 3rd round pick (... NaN  NaN  
1808  Tampa Bay Lightning Acquire:Philippe Myers · $... NaN  NaN  
1809                                                NaN NaN  NaN  

[1810 rows x 7 columns]

In [52]:
raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1810 entries, 0 to 1809
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  1810 non-null   int64  
 1   0           1810 non-null   object 
 2   1           1790 non-null   object 
 3   2           0 non-null      float64
 4   3           1754 non-null   object 
 5   4           0 non-null      float64
 6   5           13 non-null     object 
dtypes: float64(2), int64(1), object(4)
memory usage: 99.1+ KB


Initially we see more than 1800 rows, but the data is messy - 20+ rows contain all null values, 2 columns contain nothing, and column 5 only contains 13 values.  We will look further into some of these and drop the valueless columns.

In [53]:
raw['5'].value_counts()

5
Los Angeles Kings Acquire:From CAR:Justin Williams                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    1
Nashville Predators Acquire:From ARI:Stefan Elliott · $650,000$650,000$650,000Sum: $650,000$650,000$650,000Change: +$650,000Change: -$16,667Change: -$150,000                                                                                                                                                                                                                                                                                                                 

Column 5 only contains 13 rows, but the information contained is quite important as it contains cases of 3 team trades - while not common, these trades often contain prominent players.

In [54]:
raw = raw.drop(['Unnamed: 0','2','4'], axis=1)
raw

0                                                  1  \
0                 DATE                                      TRADE DETAILS   
1                 DATE                                      TRADE DETAILS   
2        Jun. 30, 2006  Columbus Blue Jackets Acquire:Freddy ModinFred...   
3        Jun. 26, 2006  Arizona Coyotes Acquire:Nick Boynton2007 4th r...   
4        Jun. 24, 2006                  Boston Bruins Acquire:Tuukka Rask   
...                ...                                                ...   
1805      Jul. 7, 2022  Montreal Canadiens Acquire:2022 1st round pick...   
1806      Jul. 7, 2022  Chicago Blackhawks Acquire:2022 1st round pick...   
1807      Jul. 7, 2022  Colorado Avalanche Acquire:Alexandar Georgiev ...   
1808      Jul. 3, 2022  Nashville Predators Acquire:Ryan McDonagh · $6...   
1809  No Results Found                                                NaN   

                                                      3    5  
0                                                   NaN  NaN  
1                                                   NaN  NaN  
2                Tampa Bay Lightning Acquire:Marc Denis  NaN  
3     Boston Bruins Acquire:Paul Mara2007 3rd round ...  NaN  
4           Toronto Maple Leafs Acquire:Andrew Raycroft  NaN  
...                                                 ...  ...  
1805  New York Islanders Acquire:Alexander Romanov ·...  NaN  
1806  Ottawa Senators Acquire:Alex DeBrincat · $6,40...  NaN  
1807  New York Rangers Acquire:2022 3rd round pick (...  NaN  
1808  Tampa Bay Lightning Acquire:Philippe Myers · $...  NaN  
1809                                                NaN  NaN  

[1810 rows x 4 columns]

Next, we will rename some columns for clarity, and also drop the rows that offer no value - index 0 and 1, plus duplicate 'header' rows from each individual webpage we scraped, as well as the 'no results found' from the additional 'load button' clicks on the webpages.

In [55]:
raw = raw.rename(columns={"0": "Date", "1": "Trade_Details_A",'3':'Trade_Details_B','5':'TradeDetails_C'})

In [56]:
raw.drop([0,1])

Date                                    Trade_Details_A  \
2        Jun. 30, 2006  Columbus Blue Jackets Acquire:Freddy ModinFred...   
3        Jun. 26, 2006  Arizona Coyotes Acquire:Nick Boynton2007 4th r...   
4        Jun. 24, 2006                  Boston Bruins Acquire:Tuukka Rask   
5        Jun. 24, 2006  Arizona Coyotes Acquire:2006 7th round pick (B...   
6        Jun. 24, 2006   Dallas Stars Acquire:Jaroslav ModryPatrik Stefan   
...                ...                                                ...   
1805      Jul. 7, 2022  Montreal Canadiens Acquire:2022 1st round pick...   
1806      Jul. 7, 2022  Chicago Blackhawks Acquire:2022 1st round pick...   
1807      Jul. 7, 2022  Colorado Avalanche Acquire:Alexandar Georgiev ...   
1808      Jul. 3, 2022  Nashville Predators Acquire:Ryan McDonagh · $6...   
1809  No Results Found                                                NaN   

                                        Trade_Details_B TradeDetails_C  
2                Tampa Bay Lightning Acquire:Marc Denis            NaN  
3     Boston Bruins Acquire:Paul Mara2007 3rd round ...            NaN  
4           Toronto Maple Leafs Acquire:Andrew Raycroft            NaN  
5     Toronto Maple Leafs Acquire:2006 6th round pic...            NaN  
6     Atlanta Thrashers Acquire:Niko Kapanen2006 7th...            NaN  
...                                                 ...            ...  
1805  New York Islanders Acquire:Alexander Romanov ·...            NaN  
1806  Ottawa Senators Acquire:Alex DeBrincat · $6,40...            NaN  
1807  New York Rangers Acquire:2022 3rd round pick (...            NaN  
1808  Tampa Bay Lightning Acquire:Philippe Myers · $...            NaN  
1809                                                NaN            NaN  

[1808 rows x 4 columns]

In [57]:
raw = raw.drop_duplicates(keep=False)
raw

Date                                    Trade_Details_A  \
2     Jun. 30, 2006  Columbus Blue Jackets Acquire:Freddy ModinFred...   
3     Jun. 26, 2006  Arizona Coyotes Acquire:Nick Boynton2007 4th r...   
4     Jun. 24, 2006                  Boston Bruins Acquire:Tuukka Rask   
5     Jun. 24, 2006  Arizona Coyotes Acquire:2006 7th round pick (B...   
6     Jun. 24, 2006   Dallas Stars Acquire:Jaroslav ModryPatrik Stefan   
...             ...                                                ...   
1804   Jul. 7, 2022  Chicago Blackhawks Acquire:2022 1st round pick...   
1805   Jul. 7, 2022  Montreal Canadiens Acquire:2022 1st round pick...   
1806   Jul. 7, 2022  Chicago Blackhawks Acquire:2022 1st round pick...   
1807   Jul. 7, 2022  Colorado Avalanche Acquire:Alexandar Georgiev ...   
1808   Jul. 3, 2022  Nashville Predators Acquire:Ryan McDonagh · $6...   

                                        Trade_Details_B TradeDetails_C  
2                Tampa Bay Lightning Acquire:Marc Denis            NaN  
3     Boston Bruins Acquire:Paul Mara2007 3rd round ...            NaN  
4           Toronto Maple Leafs Acquire:Andrew Raycroft            NaN  
5     Toronto Maple Leafs Acquire:2006 6th round pic...            NaN  
6     Atlanta Thrashers Acquire:Niko Kapanen2006 7th...            NaN  
...                                                 ...            ...  
1804  Montreal Canadiens Acquire:Kirby Dach · $0$0$0...            NaN  
1805  New York Islanders Acquire:Alexander Romanov ·...            NaN  
1806  Ottawa Senators Acquire:Alex DeBrincat · $6,40...            NaN  
1807  New York Rangers Acquire:2022 3rd round pick (...            NaN  
1808  Tampa Bay Lightning Acquire:Philippe Myers · $...            NaN  

[1754 rows x 4 columns]

After doing this we find 1754 rows containing more substance.  But we still need to dig a bit further to have it ready for further action.

In [58]:
raw['Date'].unique()

array(['Jun. 30, 2006', 'Jun. 26, 2006', 'Jun. 24, 2006', 'Jun. 23, 2006',
       'Jun. 15, 2006', 'Jun. 14, 2006', 'Jun. 1, 2006', 'May 26, 2006',
       'Mar. 9, 2006', 'Mar. 8, 2006', 'Mar. 1, 2006', 'Feb. 28, 2006',
       'Feb. 8, 2006', 'Feb. 5, 2006', 'Feb. 1, 2006', 'Jan. 30, 2006',
       'Jan. 29, 2006', 'Jan. 26, 2006', 'Jan. 24, 2006', 'Jan. 23, 2006',
       'Jan. 20, 2006', 'Jan. 18, 2006', 'Jan. 10, 2006', 'Jan. 8, 2006',
       'Jan. 4, 2006', 'Dec. 30, 2005', 'Dec. 28, 2005', 'Dec. 13, 2005',
       'Dec. 12, 2005', 'Dec. 9, 2005', 'Dec. 5, 2005', 'Dec. 2, 2005',
       'Nov. 30, 2005', 'Nov. 25, 2005', 'Nov. 19, 2005', 'Nov. 18, 2005',
       'Nov. 15, 2005', 'Nov. 6, 2005', 'Oct. 23, 2005', 'Oct. 18, 2005',
       'Oct. 8, 2005', 'Oct. 7, 2005', 'Oct. 5, 2005', 'Oct. 4, 2005',
       'Oct. 3, 2005', 'Sep. 30, 2005', 'Sep. 26, 2005', 'Sep. 9, 2005',
       'Aug. 30, 2005', 'Aug. 26, 2005', 'Aug. 25, 2005', 'Aug. 23, 2005',
       'Aug. 22, 2005', 'Aug. 10, 2005', 'Aug

Next we will split the trade details for each column into separate columns, separating the teams and 'assets' acquired.  We will also drop the intial columns after confirming the split works properly.  Lastly in this section we will rename the columns to more accurately describe the asset columns (Team_A_Acquires rather than Trade_Details_A).

In [59]:
raw['Trade_Details_A']

2       Columbus Blue Jackets Acquire:Freddy ModinFred...
3       Arizona Coyotes Acquire:Nick Boynton2007 4th r...
4                       Boston Bruins Acquire:Tuukka Rask
5       Arizona Coyotes Acquire:2006 7th round pick (B...
6        Dallas Stars Acquire:Jaroslav ModryPatrik Stefan
                              ...                        
1804    Chicago Blackhawks Acquire:2022 1st round pick...
1805    Montreal Canadiens Acquire:2022 1st round pick...
1806    Chicago Blackhawks Acquire:2022 1st round pick...
1807    Colorado Avalanche Acquire:Alexandar Georgiev ...
1808    Nashville Predators Acquire:Ryan McDonagh · $6...
Name: Trade_Details_A, Length: 1754, dtype: object

In [64]:
raw[['Team_A', 'Details_A']] = raw['Trade_Details_A'].str.split('Acquire:',n=1, expand=True)
raw

Date                                    Trade_Details_A  \
2     Jun. 30, 2006  Columbus Blue Jackets Acquire:Freddy ModinFred...   
3     Jun. 26, 2006  Arizona Coyotes Acquire:Nick Boynton2007 4th r...   
4     Jun. 24, 2006                  Boston Bruins Acquire:Tuukka Rask   
5     Jun. 24, 2006  Arizona Coyotes Acquire:2006 7th round pick (B...   
6     Jun. 24, 2006   Dallas Stars Acquire:Jaroslav ModryPatrik Stefan   
...             ...                                                ...   
1804   Jul. 7, 2022  Chicago Blackhawks Acquire:2022 1st round pick...   
1805   Jul. 7, 2022  Montreal Canadiens Acquire:2022 1st round pick...   
1806   Jul. 7, 2022  Chicago Blackhawks Acquire:2022 1st round pick...   
1807   Jul. 7, 2022  Colorado Avalanche Acquire:Alexandar Georgiev ...   
1808   Jul. 3, 2022  Nashville Predators Acquire:Ryan McDonagh · $6...   

                                        Trade_Details_B TradeDetails_C  \
2                Tampa Bay Lightning Acquire:Marc Denis            NaN   
3     Boston Bruins Acquire:Paul Mara2007 3rd round ...            NaN   
4           Toronto Maple Leafs Acquire:Andrew Raycroft            NaN   
5     Toronto Maple Leafs Acquire:2006 6th round pic...            NaN   
6     Atlanta Thrashers Acquire:Niko Kapanen2006 7th...            NaN   
...                                                 ...            ...   
1804  Montreal Canadiens Acquire:Kirby Dach · $0$0$0...            NaN   
1805  New York Islanders Acquire:Alexander Romanov ·...            NaN   
1806  Ottawa Senators Acquire:Alex DeBrincat · $6,40...            NaN   
1807  New York Rangers Acquire:2022 3rd round pick (...            NaN   
1808  Tampa Bay Lightning Acquire:Philippe Myers · $...            NaN   

                      Team_A  \
2     Columbus Blue Jackets    
3           Arizona Coyotes    
4             Boston Bruins    
5           Arizona Coyotes    
6              Dallas Stars    
...                      ...   
1804     Chicago Blackhawks    
1805     Montreal Canadiens    
1806     Chicago Blackhawks    
1807     Colorado Avalanche    
1808    Nashville Predators    

                                              Details_A  \
2                           Freddy ModinFredrik Norrena   
3     Nick Boynton2007 4th round pick (BOS - #99 - M...   
4                                           Tuukka Rask   
5     2006 7th round pick (BOS - #188 - Chris Frank)...   
6                           Jaroslav ModryPatrik Stefan   
...                                                 ...   
1804  2022 1st round pick (NYI - #13 - Frank Nazar)2...   
1805  2022 1st round pick (NYI - #13 - Frank Nazar)S...   
1806  2022 1st round pick (OTT - #7 - Kevin Korchins...   
1807  Alexandar Georgiev · $0$0$0 (Signing Rights)Su...   
1808  Ryan McDonagh · $6,750,000$6,750,000$5,460,000...   

                           Team_B  \
2     Tampa Bay Lightning Acquire   
3           Boston Bruins Acquire   
4     Toronto Maple Leafs Acquire   
5     Toronto Maple Leafs Acquire   
6       Atlanta Thrashers Acquire   
...                           ...   
1804   Montreal Canadiens Acquire   
1805   New York Islanders Acquire   
1806      Ottawa Senators Acquire   
1807     New York Rangers Acquire   
1808  Tampa Bay Lightning Acquire   

                                              Details_B Team_C Details_C  
2                                            Marc Denis    NaN       NaN  
3     Paul Mara2007 3rd round pick (ARI - #63 - Maxi...    NaN       NaN  
4                                       Andrew Raycroft    NaN       NaN  
5     2006 6th round pick (ARI - #161 - Viktor Stalb...    NaN       NaN  
6     Niko Kapanen2006 7th round pick (DAL - #210 - ...    NaN       NaN  
...                                                 ...    ...       ...  
1804  Kirby Dach · $0$0$0 (Signing Rights)Sum: $0$0$...    NaN       NaN  
1805  Alexander Romanov · $0$0$0 (Signing Rights)202...    NaN       NaN  
1806  Alex DeBrincat · 

In [65]:
raw[['Team_B', 'Details_B']] = raw['Trade_Details_B'].str.split('Acquire:',n=1, expand=True)
raw

Date                                    Trade_Details_A  \
2     Jun. 30, 2006  Columbus Blue Jackets Acquire:Freddy ModinFred...   
3     Jun. 26, 2006  Arizona Coyotes Acquire:Nick Boynton2007 4th r...   
4     Jun. 24, 2006                  Boston Bruins Acquire:Tuukka Rask   
5     Jun. 24, 2006  Arizona Coyotes Acquire:2006 7th round pick (B...   
6     Jun. 24, 2006   Dallas Stars Acquire:Jaroslav ModryPatrik Stefan   
...             ...                                                ...   
1804   Jul. 7, 2022  Chicago Blackhawks Acquire:2022 1st round pick...   
1805   Jul. 7, 2022  Montreal Canadiens Acquire:2022 1st round pick...   
1806   Jul. 7, 2022  Chicago Blackhawks Acquire:2022 1st round pick...   
1807   Jul. 7, 2022  Colorado Avalanche Acquire:Alexandar Georgiev ...   
1808   Jul. 3, 2022  Nashville Predators Acquire:Ryan McDonagh · $6...   

                                        Trade_Details_B TradeDetails_C  \
2                Tampa Bay Lightning Acquire:Marc Denis            NaN   
3     Boston Bruins Acquire:Paul Mara2007 3rd round ...            NaN   
4           Toronto Maple Leafs Acquire:Andrew Raycroft            NaN   
5     Toronto Maple Leafs Acquire:2006 6th round pic...            NaN   
6     Atlanta Thrashers Acquire:Niko Kapanen2006 7th...            NaN   
...                                                 ...            ...   
1804  Montreal Canadiens Acquire:Kirby Dach · $0$0$0...            NaN   
1805  New York Islanders Acquire:Alexander Romanov ·...            NaN   
1806  Ottawa Senators Acquire:Alex DeBrincat · $6,40...            NaN   
1807  New York Rangers Acquire:2022 3rd round pick (...            NaN   
1808  Tampa Bay Lightning Acquire:Philippe Myers · $...            NaN   

                      Team_A  \
2     Columbus Blue Jackets    
3           Arizona Coyotes    
4             Boston Bruins    
5           Arizona Coyotes    
6              Dallas Stars    
...                      ...   
1804     Chicago Blackhawks    
1805     Montreal Canadiens    
1806     Chicago Blackhawks    
1807     Colorado Avalanche    
1808    Nashville Predators    

                                              Details_A                Team_B  \
2                           Freddy ModinFredrik Norrena  Tampa Bay Lightning    
3     Nick Boynton2007 4th round pick (BOS - #99 - M...        Boston Bruins    
4                                           Tuukka Rask  Toronto Maple Leafs    
5     2006 7th round pick (BOS - #188 - Chris Frank)...  Toronto Maple Leafs    
6                           Jaroslav ModryPatrik Stefan    Atlanta Thrashers    
...                                                 ...                   ...   
1804  2022 1st round pick (NYI - #13 - Frank Nazar)2...   Montreal Canadiens    
1805  2022 1st round pick (NYI - #13 - Frank Nazar)S...   New York Islanders    
1806  2022 1st round pick (OTT - #7 - Kevin Korchins...      Ottawa Senators    
1807  Alexandar Georgiev · $0$0$0 (Signing Rights)Su...     New York Rangers    
1808  Ryan McDonagh · $6,750,000$6,750,000$5,460,000...  Tampa Bay Lightning    

                                              Details_B Team_C Details_C  
2                                            Marc Denis    NaN       NaN  
3     Paul Mara2007 3rd round pick (ARI - #63 - Maxi...    NaN       NaN  
4                                       Andrew Raycroft    NaN       NaN  
5     2006 6th round pick (ARI - #161 - Viktor Stalb...    NaN       NaN  
6     Niko Kapanen2006 7th round pick (DAL - #210 - ...    NaN       NaN  
...                                                 ...    ...       ...  
1804  Kirby Dach · $0$0$0 (Signing Rights)Sum: $0$0$...    NaN       NaN  
1805  Alexander Romanov · $0$0$0 (Signing Rights)202...    NaN       NaN  
1806  Alex DeBrincat · $6,400,000$6,400,000$9,000,00...    NaN       NaN  
1807  2022 3rd round pick (COL - #97 - Bryce McConne...    NaN       NaN  
1808  Philippe Myers · $2,550,000$2,550,000$3,800,00..

In [66]:
raw[['Team_C', 'Details_C']] = raw['TradeDetails_C'].str.split('Acquire:',n=1, expand=True)
raw

Date                                    Trade_Details_A  \
2     Jun. 30, 2006  Columbus Blue Jackets Acquire:Freddy ModinFred...   
3     Jun. 26, 2006  Arizona Coyotes Acquire:Nick Boynton2007 4th r...   
4     Jun. 24, 2006                  Boston Bruins Acquire:Tuukka Rask   
5     Jun. 24, 2006  Arizona Coyotes Acquire:2006 7th round pick (B...   
6     Jun. 24, 2006   Dallas Stars Acquire:Jaroslav ModryPatrik Stefan   
...             ...                                                ...   
1804   Jul. 7, 2022  Chicago Blackhawks Acquire:2022 1st round pick...   
1805   Jul. 7, 2022  Montreal Canadiens Acquire:2022 1st round pick...   
1806   Jul. 7, 2022  Chicago Blackhawks Acquire:2022 1st round pick...   
1807   Jul. 7, 2022  Colorado Avalanche Acquire:Alexandar Georgiev ...   
1808   Jul. 3, 2022  Nashville Predators Acquire:Ryan McDonagh · $6...   

                                        Trade_Details_B TradeDetails_C  \
2                Tampa Bay Lightning Acquire:Marc Denis            NaN   
3     Boston Bruins Acquire:Paul Mara2007 3rd round ...            NaN   
4           Toronto Maple Leafs Acquire:Andrew Raycroft            NaN   
5     Toronto Maple Leafs Acquire:2006 6th round pic...            NaN   
6     Atlanta Thrashers Acquire:Niko Kapanen2006 7th...            NaN   
...                                                 ...            ...   
1804  Montreal Canadiens Acquire:Kirby Dach · $0$0$0...            NaN   
1805  New York Islanders Acquire:Alexander Romanov ·...            NaN   
1806  Ottawa Senators Acquire:Alex DeBrincat · $6,40...            NaN   
1807  New York Rangers Acquire:2022 3rd round pick (...            NaN   
1808  Tampa Bay Lightning Acquire:Philippe Myers · $...            NaN   

                      Team_A  \
2     Columbus Blue Jackets    
3           Arizona Coyotes    
4             Boston Bruins    
5           Arizona Coyotes    
6              Dallas Stars    
...                      ...   
1804     Chicago Blackhawks    
1805     Montreal Canadiens    
1806     Chicago Blackhawks    
1807     Colorado Avalanche    
1808    Nashville Predators    

                                              Details_A                Team_B  \
2                           Freddy ModinFredrik Norrena  Tampa Bay Lightning    
3     Nick Boynton2007 4th round pick (BOS - #99 - M...        Boston Bruins    
4                                           Tuukka Rask  Toronto Maple Leafs    
5     2006 7th round pick (BOS - #188 - Chris Frank)...  Toronto Maple Leafs    
6                           Jaroslav ModryPatrik Stefan    Atlanta Thrashers    
...                                                 ...                   ...   
1804  2022 1st round pick (NYI - #13 - Frank Nazar)2...   Montreal Canadiens    
1805  2022 1st round pick (NYI - #13 - Frank Nazar)S...   New York Islanders    
1806  2022 1st round pick (OTT - #7 - Kevin Korchins...      Ottawa Senators    
1807  Alexandar Georgiev · $0$0$0 (Signing Rights)Su...     New York Rangers    
1808  Ryan McDonagh · $6,750,000$6,750,000$5,460,000...  Tampa Bay Lightning    

                                              Details_B Team_C Details_C  
2                                            Marc Denis    NaN       NaN  
3     Paul Mara2007 3rd round pick (ARI - #63 - Maxi...    NaN       NaN  
4                                       Andrew Raycroft    NaN       NaN  
5     2006 6th round pick (ARI - #161 - Viktor Stalb...    NaN       NaN  
6     Niko Kapanen2006 7th round pick (DAL - #210 - ...    NaN       NaN  
...                                                 ...    ...       ...  
1804  Kirby Dach · $0$0$0 (Signing Rights)Sum: $0$0$...    NaN       NaN  
1805  Alexander Romanov · $0$0$0 (Signing Rights)202...    NaN       NaN  
1806  Alex DeBrincat · $6,400,000$6,400,000$9,000,00...    NaN       NaN  
1807  2022 3rd round pick (COL - #97 - Bryce McConne...    NaN       NaN  
1808  Philippe Myers · $2,550,000$2,550,000$3,800,00..

In [67]:
raw = raw.drop(['Trade_Details_A','Trade_Details_B','TradeDetails_C'], axis=1)
raw

Date                  Team_A  \
2     Jun. 30, 2006  Columbus Blue Jackets    
3     Jun. 26, 2006        Arizona Coyotes    
4     Jun. 24, 2006          Boston Bruins    
5     Jun. 24, 2006        Arizona Coyotes    
6     Jun. 24, 2006           Dallas Stars    
...             ...                     ...   
1804   Jul. 7, 2022     Chicago Blackhawks    
1805   Jul. 7, 2022     Montreal Canadiens    
1806   Jul. 7, 2022     Chicago Blackhawks    
1807   Jul. 7, 2022     Colorado Avalanche    
1808   Jul. 3, 2022    Nashville Predators    

                                              Details_A                Team_B  \
2                           Freddy ModinFredrik Norrena  Tampa Bay Lightning    
3     Nick Boynton2007 4th round pick (BOS - #99 - M...        Boston Bruins    
4                                           Tuukka Rask  Toronto Maple Leafs    
5     2006 7th round pick (BOS - #188 - Chris Frank)...  Toronto Maple Leafs    
6                           Jaroslav ModryPatrik Stefan    Atlanta Thrashers    
...                                                 ...                   ...   
1804  2022 1st round pick (NYI - #13 - Frank Nazar)2...   Montreal Canadiens    
1805  2022 1st round pick (NYI - #13 - Frank Nazar)S...   New York Islanders    
1806  2022 1st round pick (OTT - #7 - Kevin Korchins...      Ottawa Senators    
1807  Alexandar Georgiev · $0$0$0 (Signing Rights)Su...     New York Rangers    
1808  Ryan McDonagh · $6,750,000$6,750,000$5,460,000...  Tampa Bay Lightning    

                                              Details_B Team_C Details_C  
2                                            Marc Denis    NaN       NaN  
3     Paul Mara2007 3rd round pick (ARI - #63 - Maxi...    NaN       NaN  
4                                       Andrew Raycroft    NaN       NaN  
5     2006 6th round pick (ARI - #161 - Viktor Stalb...    NaN       NaN  
6     Niko Kapanen2006 7th round pick (DAL - #210 - ...    NaN       NaN  
...                                                 ...    ...       ...  
1804  Kirby Dach · $0$0$0 (Signing Rights)Sum: $0$0$...    NaN       NaN  
1805  Alexander Romanov · $0$0$0 (Signing Rights)202...    NaN       NaN  
1806  Alex DeBrincat · $6,400,000$6,400,000$9,000,00...    NaN       NaN  
1807  2022 3rd round pick (COL - #97 - Bryce McConne...    NaN       NaN  
1808  Philippe Myers · $2,550,000$2,550,000$3,800,00...    NaN       NaN  

[1754 rows x 7 columns]

In [68]:
raw = raw.rename(columns={"Details_A": "Team_A_Acquires", "Details_B": "Team_B_Acquires",'Details_C':'Team_C_Acquires'})
raw

Date                  Team_A  \
2     Jun. 30, 2006  Columbus Blue Jackets    
3     Jun. 26, 2006        Arizona Coyotes    
4     Jun. 24, 2006          Boston Bruins    
5     Jun. 24, 2006        Arizona Coyotes    
6     Jun. 24, 2006           Dallas Stars    
...             ...                     ...   
1804   Jul. 7, 2022     Chicago Blackhawks    
1805   Jul. 7, 2022     Montreal Canadiens    
1806   Jul. 7, 2022     Chicago Blackhawks    
1807   Jul. 7, 2022     Colorado Avalanche    
1808   Jul. 3, 2022    Nashville Predators    

                                        Team_A_Acquires                Team_B  \
2                           Freddy ModinFredrik Norrena  Tampa Bay Lightning    
3     Nick Boynton2007 4th round pick (BOS - #99 - M...        Boston Bruins    
4                                           Tuukka Rask  Toronto Maple Leafs    
5     2006 7th round pick (BOS - #188 - Chris Frank)...  Toronto Maple Leafs    
6                           Jaroslav ModryPatrik Stefan    Atlanta Thrashers    
...                                                 ...                   ...   
1804  2022 1st round pick (NYI - #13 - Frank Nazar)2...   Montreal Canadiens    
1805  2022 1st round pick (NYI - #13 - Frank Nazar)S...   New York Islanders    
1806  2022 1st round pick (OTT - #7 - Kevin Korchins...      Ottawa Senators    
1807  Alexandar Georgiev · $0$0$0 (Signing Rights)Su...     New York Rangers    
1808  Ryan McDonagh · $6,750,000$6,750,000$5,460,000...  Tampa Bay Lightning    

                                        Team_B_Acquires Team_C Team_C_Acquires  
2                                            Marc Denis    NaN             NaN  
3     Paul Mara2007 3rd round pick (ARI - #63 - Maxi...    NaN             NaN  
4                                       Andrew Raycroft    NaN             NaN  
5     2006 6th round pick (ARI - #161 - Viktor Stalb...    NaN             NaN  
6     Niko Kapanen2006 7th round pick (DAL - #210 - ...    NaN             NaN  
...                                                 ...    ...             ...  
1804  Kirby Dach · $0$0$0 (Signing Rights)Sum: $0$0$...    NaN             NaN  
1805  Alexander Romanov · $0$0$0 (Signing Rights)202...    NaN             NaN  
1806  Alex DeBrincat · $6,400,000$6,400,000$9,000,00...    NaN             NaN  
1807  2022 3rd round pick (COL - #97 - Bryce McConne...    NaN             NaN  
1808  Philippe Myers · $2,550,000$2,550,000$3,800,00...    NaN             NaN  

[1754 rows x 7 columns]

Additionally we need to convert to datetime in order to parse through the dates that will constitute a 'midseason' trade for each year.

In [72]:
raw['Date'] = pd.to_datetime(raw['Date'], format='mixed')
raw

Date                  Team_A  \
2    2006-06-30  Columbus Blue Jackets    
3    2006-06-26        Arizona Coyotes    
4    2006-06-24          Boston Bruins    
5    2006-06-24        Arizona Coyotes    
6    2006-06-24           Dallas Stars    
...         ...                     ...   
1804 2022-07-07     Chicago Blackhawks    
1805 2022-07-07     Montreal Canadiens    
1806 2022-07-07     Chicago Blackhawks    
1807 2022-07-07     Colorado Avalanche    
1808 2022-07-03    Nashville Predators    

                                        Team_A_Acquires                Team_B  \
2                           Freddy ModinFredrik Norrena  Tampa Bay Lightning    
3     Nick Boynton2007 4th round pick (BOS - #99 - M...        Boston Bruins    
4                                           Tuukka Rask  Toronto Maple Leafs    
5     2006 7th round pick (BOS - #188 - Chris Frank)...  Toronto Maple Leafs    
6                           Jaroslav ModryPatrik Stefan    Atlanta Thrashers    
...                                                 ...                   ...   
1804  2022 1st round pick (NYI - #13 - Frank Nazar)2...   Montreal Canadiens    
1805  2022 1st round pick (NYI - #13 - Frank Nazar)S...   New York Islanders    
1806  2022 1st round pick (OTT - #7 - Kevin Korchins...      Ottawa Senators    
1807  Alexandar Georgiev · $0$0$0 (Signing Rights)Su...     New York Rangers    
1808  Ryan McDonagh · $6,750,000$6,750,000$5,460,000...  Tampa Bay Lightning    

                                        Team_B_Acquires Team_C Team_C_Acquires  
2                                            Marc Denis    NaN             NaN  
3     Paul Mara2007 3rd round pick (ARI - #63 - Maxi...    NaN             NaN  
4                                       Andrew Raycroft    NaN             NaN  
5     2006 6th round pick (ARI - #161 - Viktor Stalb...    NaN             NaN  
6     Niko Kapanen2006 7th round pick (DAL - #210 - ...    NaN             NaN  
...                                                 ...    ...             ...  
1804  Kirby Dach · $0$0$0 (Signing Rights)Sum: $0$0$...    NaN             NaN  
1805  Alexander Romanov · $0$0$0 (Signing Rights)202...    NaN             NaN  
1806  Alex DeBrincat · $6,400,000$6,400,000$9,000,00...    NaN             NaN  
1807  2022 3rd round pick (COL - #97 - Bryce McConne...    NaN             NaN  
1808  Philippe Myers · $2,550,000$2,550,000$3,800,00...    NaN             NaN  

[1754 rows x 7 columns]

Further down in the notebook, we learn that we are actually missing a few years of data from the initial scrape.  After doing some more work in the webscraping notebook, we have this additional CSV.  We will repeat the steps on 'raw' for 'raw2' before proceeding.

In [127]:
raw2 = pd.read_csv('/Users/phil/Documents/Capstone/Data/test_df_2.csv')
raw2

Unnamed: 0              0  \
0             0           DATE   
1             1           DATE   
2             2  Jun. 27, 2012   
3             3  Jun. 23, 2012   
4             4  Jun. 23, 2012   
..          ...            ...   
339          69  Aug. 18, 2023   
340          70  Aug. 15, 2023   
341          71   Aug. 9, 2023   
342          72   Aug. 6, 2023   
343          73   Jul. 9, 2023   

                                                     1   2  \
0                                        TRADE DETAILS NaN   
1                                        TRADE DETAILS NaN   
2    Calgary Flames Acquire:Dennis Wideman · $5,250... NaN   
3    Philadelphia Flyers Acquire:Luke Schenn · $3,6... NaN   
4    Boston Bruins Acquire:Michel Ouellet · $0$0$0 ... NaN   
..                                                 ...  ..   
339  Anaheim Ducks Acquire:Ilya Lyubushkin · $2,750... NaN   
340  Detroit Red Wings Acquire:Jeff Petry ($2,343,7... NaN   
341  Carolina Hurricanes Acquire:David Kase · $0$0$... NaN   
342  Montreal Canadiens Acquire:From PIT:Jeff Petry... NaN   
343  Detroit Red Wings Acquire:Alex DeBrincat · $0$... NaN   

                                                     3   4  \
0                                                  NaN NaN   
1                                                  NaN NaN   
2    Washington Capitals Acquire:Jordan Henry · $0$... NaN   
3    Toronto Maple Leafs Acquire:James van Riemsdyk... NaN   
4    Tampa Bay Lightning Acquire:Benoit Pouliot · $... NaN   
..                                                 ...  ..   
339  Buffalo Sabres Acquire:2025 4th round pick (MI... NaN   
340  Montreal Canadiens Acquire:Gustav Lindström · ... NaN   
341  Philadelphia Flyers Acquire:Massimo Rizzo · $0... NaN   
342  Pittsburgh Penguins Acquire:From SJS:Erik Karl... NaN   
343  Ottawa Senators Acquire:Dominik Kubalik · $2,5... NaN   

                                                     5  
0                                                  NaN  
1                                                  NaN  
2                                                  NaN  
3                                                  NaN  
4                                                  NaN  
..                                                 ...  
339                                                NaN  
340                                                NaN  
341                                                NaN  
342  San Jose Sharks Acquire:From MTL:Mike Hoffman ...  
343                                                NaN  

[344 rows x 7 columns]

In [128]:
raw2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  344 non-null    int64  
 1   0           344 non-null    object 
 2   1           344 non-null    object 
 3   2           0 non-null      float64
 4   3           336 non-null    object 
 5   4           0 non-null      float64
 6   5           8 non-null      object 
dtypes: float64(2), int64(1), object(4)
memory usage: 18.9+ KB


In [129]:
raw2 = raw2.drop(['Unnamed: 0','2','4'], axis=1)
raw2

0                                                  1  \
0             DATE                                      TRADE DETAILS   
1             DATE                                      TRADE DETAILS   
2    Jun. 27, 2012  Calgary Flames Acquire:Dennis Wideman · $5,250...   
3    Jun. 23, 2012  Philadelphia Flyers Acquire:Luke Schenn · $3,6...   
4    Jun. 23, 2012  Boston Bruins Acquire:Michel Ouellet · $0$0$0 ...   
..             ...                                                ...   
339  Aug. 18, 2023  Anaheim Ducks Acquire:Ilya Lyubushkin · $2,750...   
340  Aug. 15, 2023  Detroit Red Wings Acquire:Jeff Petry ($2,343,7...   
341   Aug. 9, 2023  Carolina Hurricanes Acquire:David Kase · $0$0$...   
342   Aug. 6, 2023  Montreal Canadiens Acquire:From PIT:Jeff Petry...   
343   Jul. 9, 2023  Detroit Red Wings Acquire:Alex DeBrincat · $0$...   

                                                     3  \
0                                                  NaN   
1                                                  NaN   
2    Washington Capitals Acquire:Jordan Henry · $0$...   
3    Toronto Maple Leafs Acquire:James van Riemsdyk...   
4    Tampa Bay Lightning Acquire:Benoit Pouliot · $...   
..                                                 ...   
339  Buffalo Sabres Acquire:2025 4th round pick (MI...   
340  Montreal Canadiens Acquire:Gustav Lindström · ...   
341  Philadelphia Flyers Acquire:Massimo Rizzo · $0...   
342  Pittsburgh Penguins Acquire:From SJS:Erik Karl...   
343  Ottawa Senators Acquire:Dominik Kubalik · $2,5...   

                                                     5  
0                                                  NaN  
1                                                  NaN  
2                                                  NaN  
3                                                  NaN  
4                                                  NaN  
..                                                 ...  
339                                                NaN  
340                                                NaN  
341                                                NaN  
342  San Jose Sharks Acquire:From MTL:Mike Hoffman ...  
343                                                NaN  

[344 rows x 4 columns]

In [130]:
raw2 = raw2.rename(columns={"0": "Date", "1": "Trade_Details_A",'3':'Trade_Details_B','5':'TradeDetails_C'})

In [131]:
raw2 = raw2.drop_duplicates(keep=False)
raw2

Date                                    Trade_Details_A  \
2    Jun. 27, 2012  Calgary Flames Acquire:Dennis Wideman · $5,250...   
3    Jun. 23, 2012  Philadelphia Flyers Acquire:Luke Schenn · $3,6...   
4    Jun. 23, 2012  Boston Bruins Acquire:Michel Ouellet · $0$0$0 ...   
5    Jun. 23, 2012  Chicago Blackhawks Acquire:2013 4th round pick...   
6    Jun. 23, 2012  Dallas Stars Acquire:2013 7th round pick (FLA ...   
..             ...                                                ...   
339  Aug. 18, 2023  Anaheim Ducks Acquire:Ilya Lyubushkin · $2,750...   
340  Aug. 15, 2023  Detroit Red Wings Acquire:Jeff Petry ($2,343,7...   
341   Aug. 9, 2023  Carolina Hurricanes Acquire:David Kase · $0$0$...   
342   Aug. 6, 2023  Montreal Canadiens Acquire:From PIT:Jeff Petry...   
343   Jul. 9, 2023  Detroit Red Wings Acquire:Alex DeBrincat · $0$...   

                                       Trade_Details_B  \
2    Washington Capitals Acquire:Jordan Henry · $0$...   
3    Toronto Maple Leafs Acquire:James van Riemsdyk...   
4    Tampa Bay Lightning Acquire:Benoit Pouliot · $...   
5    San Jose Sharks Acquire:2012 4th round pick (C...   
6    Florida Panthers Acquire:2012 7th round pick (...   
..                                                 ...   
339  Buffalo Sabres Acquire:2025 4th round pick (MI...   
340  Montreal Canadiens Acquire:Gustav Lindström · ...   
341  Philadelphia Flyers Acquire:Massimo Rizzo · $0...   
342  Pittsburgh Penguins Acquire:From SJS:Erik Karl...   
343  Ottawa Senators Acquire:Dominik Kubalik · $2,5...   

                                        TradeDetails_C  
2                                                  NaN  
3                                                  NaN  
4                                                  NaN  
5                                                  NaN  
6                                                  NaN  
..                                                 ...  
339                                                NaN  
340                                                NaN  
341                                                NaN  
342  San Jose Sharks Acquire:From MTL:Mike Hoffman ...  
343                                                NaN  

[336 rows x 4 columns]

In [132]:
raw2[['Team_A', 'Team_A_Acquires']] = raw2['Trade_Details_A'].str.split('Acquire:',n=1, expand=True)

In [133]:
raw2[['Team_B', 'Team_B_Acquires']] = raw2['Trade_Details_B'].str.split('Acquire:',n=1, expand=True)

In [135]:
raw2[['Team_C', 'Team_C_Acquires']] = raw2['TradeDetails_C'].str.split('Acquire:',n=1, expand=True)
raw2

Date                                    Trade_Details_A  \
2    Jun. 27, 2012  Calgary Flames Acquire:Dennis Wideman · $5,250...   
3    Jun. 23, 2012  Philadelphia Flyers Acquire:Luke Schenn · $3,6...   
4    Jun. 23, 2012  Boston Bruins Acquire:Michel Ouellet · $0$0$0 ...   
5    Jun. 23, 2012  Chicago Blackhawks Acquire:2013 4th round pick...   
6    Jun. 23, 2012  Dallas Stars Acquire:2013 7th round pick (FLA ...   
..             ...                                                ...   
339  Aug. 18, 2023  Anaheim Ducks Acquire:Ilya Lyubushkin · $2,750...   
340  Aug. 15, 2023  Detroit Red Wings Acquire:Jeff Petry ($2,343,7...   
341   Aug. 9, 2023  Carolina Hurricanes Acquire:David Kase · $0$0$...   
342   Aug. 6, 2023  Montreal Canadiens Acquire:From PIT:Jeff Petry...   
343   Jul. 9, 2023  Detroit Red Wings Acquire:Alex DeBrincat · $0$...   

                                       Trade_Details_B  \
2    Washington Capitals Acquire:Jordan Henry · $0$...   
3    Toronto Maple Leafs Acquire:James van Riemsdyk...   
4    Tampa Bay Lightning Acquire:Benoit Pouliot · $...   
5    San Jose Sharks Acquire:2012 4th round pick (C...   
6    Florida Panthers Acquire:2012 7th round pick (...   
..                                                 ...   
339  Buffalo Sabres Acquire:2025 4th round pick (MI...   
340  Montreal Canadiens Acquire:Gustav Lindström · ...   
341  Philadelphia Flyers Acquire:Massimo Rizzo · $0...   
342  Pittsburgh Penguins Acquire:From SJS:Erik Karl...   
343  Ottawa Senators Acquire:Dominik Kubalik · $2,5...   

                                        TradeDetails_C                Team_A  \
2                                                  NaN       Calgary Flames    
3                                                  NaN  Philadelphia Flyers    
4                                                  NaN        Boston Bruins    
5                                                  NaN   Chicago Blackhawks    
6                                                  NaN         Dallas Stars    
..                                                 ...                   ...   
339                                                NaN        Anaheim Ducks    
340                                                NaN    Detroit Red Wings    
341                                                NaN  Carolina Hurricanes    
342  San Jose Sharks Acquire:From MTL:Mike Hoffman ...   Montreal Canadiens    
343                                                NaN    Detroit Red Wings    

                                       Team_A_Acquires                Team_B  \
2    Dennis Wideman · $5,250,000$5,250,000$5,000,00...  Washington Capitals    
3    Luke Schenn · $3,600,000$3,600,000$3,350,000Su...  Toronto Maple Leafs    
4    Michel Ouellet · $0$0$0 (Signing Rights)2012 5...  Tampa Bay Lightning    
5    2013 4th round pick (SJS - #111 - Robin Norell...      San Jose Sharks    
6    2013 7th round pick (FLA - #182 - Aleksi Makel...     Florida Panthers    
..                                                 ...                   ...   
339  Ilya Lyubushkin · $2,750,000$2,750,000$2,750,0...       Buffalo Sabres    
340  Jeff Petry ($2,343,750 retained - 50%) · $2,34...   Montreal Canadiens    
341  David Kase · $0$0$0 (AHL/JR) (Signing Rights)S...  Philadelphia Flyers    
342  From PIT:Jeff Petry ($1,562,500 retained - 25%...  Pittsburgh Penguins    
343  Alex DeBrincat · $0$0$0 (Signing Rights)Sum: $...      Ottawa Senators    

                                       Team_B_Acquires            Team_C  \
2    Jordan Henry · $0$0$0 (Signing Rights)2013 5th...               NaN   
3    James van Riemsdyk · $4,250,000$4,250,000$2,50...               NaN   
4    Benoit Pouliot · $1,800,000$1,800,000$1,800,00...               NaN   
5    2012 4th round pick (CHI - #109 - Christophe L...               NaN   
6    2012 7th round pick (DAL - #194 - Jonatan Niel...               NaN   
..                                                 ...        

In [136]:
raw2 = raw2.drop(['Trade_Details_A','Trade_Details_B','TradeDetails_C'], axis=1)
raw2

Date                Team_A  \
2    Jun. 27, 2012       Calgary Flames    
3    Jun. 23, 2012  Philadelphia Flyers    
4    Jun. 23, 2012        Boston Bruins    
5    Jun. 23, 2012   Chicago Blackhawks    
6    Jun. 23, 2012         Dallas Stars    
..             ...                   ...   
339  Aug. 18, 2023        Anaheim Ducks    
340  Aug. 15, 2023    Detroit Red Wings    
341   Aug. 9, 2023  Carolina Hurricanes    
342   Aug. 6, 2023   Montreal Canadiens    
343   Jul. 9, 2023    Detroit Red Wings    

                                       Team_A_Acquires                Team_B  \
2    Dennis Wideman · $5,250,000$5,250,000$5,000,00...  Washington Capitals    
3    Luke Schenn · $3,600,000$3,600,000$3,350,000Su...  Toronto Maple Leafs    
4    Michel Ouellet · $0$0$0 (Signing Rights)2012 5...  Tampa Bay Lightning    
5    2013 4th round pick (SJS - #111 - Robin Norell...      San Jose Sharks    
6    2013 7th round pick (FLA - #182 - Aleksi Makel...     Florida Panthers    
..                                                 ...                   ...   
339  Ilya Lyubushkin · $2,750,000$2,750,000$2,750,0...       Buffalo Sabres    
340  Jeff Petry ($2,343,750 retained - 50%) · $2,34...   Montreal Canadiens    
341  David Kase · $0$0$0 (AHL/JR) (Signing Rights)S...  Philadelphia Flyers    
342  From PIT:Jeff Petry ($1,562,500 retained - 25%...  Pittsburgh Penguins    
343  Alex DeBrincat · $0$0$0 (Signing Rights)Sum: $...      Ottawa Senators    

                                       Team_B_Acquires            Team_C  \
2    Jordan Henry · $0$0$0 (Signing Rights)2013 5th...               NaN   
3    James van Riemsdyk · $4,250,000$4,250,000$2,50...               NaN   
4    Benoit Pouliot · $1,800,000$1,800,000$1,800,00...               NaN   
5    2012 4th round pick (CHI - #109 - Christophe L...               NaN   
6    2012 7th round pick (DAL - #194 - Jonatan Niel...               NaN   
..                                                 ...               ...   
339  2025 4th round pick (MIN)Sum: $0$0$0Change: -$...               NaN   
340  Gustav Lindström · $950,000$950,000$950,000202...               NaN   
341  Massimo Rizzo · $0$0$0 (AHL/JR) (Signing Right...               NaN   
342  From SJS:Erik Karlsson ($1,500,000 retained - ...  San Jose Sharks    
343  Dominik Kubalik · $2,500,000$2,500,000$2,500,0...               NaN   

                                       Team_C_Acquires  
2                                                  NaN  
3                                                  NaN  
4                                                  NaN  
5                                                  NaN  
6                                                  NaN  
..                                                 ...  
339                                                NaN  
340                                                NaN  
341                                                NaN  
342  From MTL:Mike Hoffman · $4,500,000$4,500,000$5...  
343                                                NaN  

[336 rows x 7 columns]

In [137]:
raw2['Date'] = pd.to_datetime(raw2['Date'], format='mixed')
raw2

Date                Team_A  \
2   2012-06-27       Calgary Flames    
3   2012-06-23  Philadelphia Flyers    
4   2012-06-23        Boston Bruins    
5   2012-06-23   Chicago Blackhawks    
6   2012-06-23         Dallas Stars    
..         ...                   ...   
339 2023-08-18        Anaheim Ducks    
340 2023-08-15    Detroit Red Wings    
341 2023-08-09  Carolina Hurricanes    
342 2023-08-06   Montreal Canadiens    
343 2023-07-09    Detroit Red Wings    

                                       Team_A_Acquires                Team_B  \
2    Dennis Wideman · $5,250,000$5,250,000$5,000,00...  Washington Capitals    
3    Luke Schenn · $3,600,000$3,600,000$3,350,000Su...  Toronto Maple Leafs    
4    Michel Ouellet · $0$0$0 (Signing Rights)2012 5...  Tampa Bay Lightning    
5    2013 4th round pick (SJS - #111 - Robin Norell...      San Jose Sharks    
6    2013 7th round pick (FLA - #182 - Aleksi Makel...     Florida Panthers    
..                                                 ...                   ...   
339  Ilya Lyubushkin · $2,750,000$2,750,000$2,750,0...       Buffalo Sabres    
340  Jeff Petry ($2,343,750 retained - 50%) · $2,34...   Montreal Canadiens    
341  David Kase · $0$0$0 (AHL/JR) (Signing Rights)S...  Philadelphia Flyers    
342  From PIT:Jeff Petry ($1,562,500 retained - 25%...  Pittsburgh Penguins    
343  Alex DeBrincat · $0$0$0 (Signing Rights)Sum: $...      Ottawa Senators    

                                       Team_B_Acquires            Team_C  \
2    Jordan Henry · $0$0$0 (Signing Rights)2013 5th...               NaN   
3    James van Riemsdyk · $4,250,000$4,250,000$2,50...               NaN   
4    Benoit Pouliot · $1,800,000$1,800,000$1,800,00...               NaN   
5    2012 4th round pick (CHI - #109 - Christophe L...               NaN   
6    2012 7th round pick (DAL - #194 - Jonatan Niel...               NaN   
..                                                 ...               ...   
339  2025 4th round pick (MIN)Sum: $0$0$0Change: -$...               NaN   
340  Gustav Lindström · $950,000$950,000$950,000202...               NaN   
341  Massimo Rizzo · $0$0$0 (AHL/JR) (Signing Right...               NaN   
342  From SJS:Erik Karlsson ($1,500,000 retained - ...  San Jose Sharks    
343  Dominik Kubalik · $2,500,000$2,500,000$2,500,0...               NaN   

                                       Team_C_Acquires  
2                                                  NaN  
3                                                  NaN  
4                                                  NaN  
5                                                  NaN  
6                                                  NaN  
..                                                 ...  
339                                                NaN  
340                                                NaN  
341                                                NaN  
342  From MTL:Mike Hoffman · $4,500,000$4,500,000$5...  
343                                                NaN  

[336 rows x 7 columns]

After cleaning up 'raw2' with the additional dates, we will go ahead and pull the 'midseason' trades for each year into their own dataframes to then be concatenated.

In [141]:
trades_06 = (raw['Date']>='2006-1-1') & (raw['Date']<='2006-3-9')
raw06 = raw.loc[trades_06]

In [142]:
trades_07 = (raw['Date']>='2007-1-1') & (raw['Date']<='2007-2-27')
raw07 = raw.loc[trades_07]

In [143]:
trades_08 = (raw['Date']>='2008-1-1') & (raw['Date']<='2008-2-28')
raw08 = raw.loc[trades_08]

In [144]:
trades_09 = (raw['Date']>='2009-1-1') & (raw['Date']<='2009-3-4')
raw09 = raw.loc[trades_09]

In [145]:
trades_10 = (raw['Date']>='2010-1-1') & (raw['Date']<='2010-3-3')
raw10 = raw.loc[trades_10]

In [146]:
trades_11 = (raw['Date']>='2011-1-1') & (raw['Date']<='2011-2-28')
raw11 = raw.loc[trades_11]

In [147]:
trades_12 = (raw2['Date']>='2012-1-1') & (raw2['Date']<='2012-2-27')
raw12 = raw2.loc[trades_12]

In [108]:
raw[raw['Date'].dt.year == 2012]

Date                  Team_A  \
743 2012-06-27         Calgary Flames    
744 2012-06-23    Philadelphia Flyers    
745 2012-06-23          Boston Bruins    
746 2012-06-23     Chicago Blackhawks    
747 2012-06-23           Dallas Stars    
748 2012-06-23           Dallas Stars    
749 2012-06-23    Nashville Predators    
750 2012-06-23    Nashville Predators    
751 2012-06-22    Toronto Maple Leafs    
752 2012-06-22        Arizona Coyotes    
753 2012-06-22           Dallas Stars    
754 2012-06-22    Carolina Hurricanes    
755 2012-06-22         Buffalo Sabres    
756 2012-06-22  Columbus Blue Jackets    
757 2012-06-22          Anaheim Ducks    
758 2012-06-15    Nashville Predators    
759 2012-06-10      Detroit Red Wings    
760 2012-06-04    Pittsburgh Penguins    
761 2012-05-26          Boston Bruins    
762 2012-03-20          Boston Bruins    
763 2012-02-27          Anaheim Ducks    
764 2012-02-27         Buffalo Sabres    
765 2012-02-27          Anaheim Ducks    
766 2012-02-27          Boston Bruins    
859 2012-07-23  Columbus Blue Jackets    
860 2012-07-20       Florida Panthers    
861 2012-07-10        St. Louis Blues    
862 2012-07-02         Buffalo Sabres    
863 2012-07-01  Columbus Blue Jackets    

                                       Team_A_Acquires                Team_B  \
743  Dennis Wideman · $5,250,000$5,250,000$5,000,00...  Washington Capitals    
744  Luke Schenn · $3,600,000$3,600,000$3,350,000Su...  Toronto Maple Leafs    
745  Michel Ouellet · $0$0$0 (Signing Rights)2012 5...  Tampa Bay Lightning    
746  2013 4th round pick (SJS - #111 - Robin Norell...      San Jose Sharks    
747  2013 7th round pick (FLA - #182 - Aleksi Makel...     Florida Panthers    
748  2012 7th round pick (EDM - #183 - Dmitry Sinit...    Los Angeles Kings    
749  2013 5th round pick (NYR - #140 - Teemu Kiviha...     New York Rangers    
750  2012 3rd round pick (NYR - #89 - Brendan Leips...     New York Rangers    
751  2013 7th round pick (WPG - #194 - Marcus Karls...        Winnipeg Jets    
752  Zbynek Michalek · $4,000,000$4,000,000$4,000,0...  Pittsburgh Penguins    
753  Cody Eakin · $571,667$571,667$550,0002012 2nd ...  Washington Capitals    
754  Jordan Staal · $4,000,000$4,000,000$4,500,000S...  Pittsburgh Penguins    
755  2012 1st round pick (CGY - #14 - Zemgus Girgen...       Calgary Flames    
756  Sergei Bobrovsky · $900,000$900,000$900,000Sum...  Philadelphia Flyers    
757  2013 2nd round pick (NYI - #45 - Nick Sörensen...   New York Islanders    
758  Sebastien Caron · $0$0$0 (Signing Rights)2013 ...  Tampa Bay Lightning    
759  Andrew Murray · $600,000$600,000$600,0002014 7...      San Jose Sharks    
760  Tomas Vokoun · $2,000,000$2,000,000$2,000,000S...  Washington Capitals    
761  Chris Bourque · $0$0$0 (Signing Rights)Sum: $0...  Washington Capitals    
762                              Future Considerations   New York Islanders    
763                                   Sebastian Erixon    Vancouver Canucks    
764                       Cody HodgsonAlexander Sulzer    Vancouver Canucks    
765                                      Dale Mitchell  Toronto Maple Leafs    
766                                         Greg Zanon       Minnesota Wild    
859  Tim Erixon · $0$900,000$900,000Artem Anisimov ...     New York Rangers    
860  Casey Wellman · $0$800,000$800,000 (AHL/JR)Sum...     New York Rangers    
861  2014 4th round pick (TBL - #110 - Austin Pogan...  Tampa Bay Lightning    
862  Adam Pardy · $2,000,000$2,000,000$2,000,000Ste...         Dallas Stars    
863  Nick Foligno · $3,083,333$3,083,333$2,500,000S...      Ottawa Senators    

                                       Team_B_Acquires Team_C Team_C_Acquires  
743  Jordan Henry · $0$0$0 (Signing Rights)2013 5th...    NaN             NaN  
744  James van Riemsdyk · $4,250,000$4,250,000$2,50...    NaN             NaN  
745  Benoit Pouliot · $1,800,000$1,800,000$1,800,00...    NaN             NaN  
746  2012 4th round pick (CH

In [124]:
raw.loc[760:780]

Date                  Team_A  \
760 2012-06-04    Pittsburgh Penguins    
761 2012-05-26          Boston Bruins    
762 2012-03-20          Boston Bruins    
763 2012-02-27          Anaheim Ducks    
764 2012-02-27         Buffalo Sabres    
765 2012-02-27          Anaheim Ducks    
766 2012-02-27          Boston Bruins    
769 2013-06-30         Minnesota Wild    
770 2013-06-30  Columbus Blue Jackets    
771 2013-06-30       Florida Panthers    
772 2013-06-30        Edmonton Oilers    
773 2013-06-30     Chicago Blackhawks    
774 2013-06-30    Washington Capitals    
775 2013-06-30         Minnesota Wild    
776 2013-06-30        Arizona Coyotes    
777 2013-06-30    Pittsburgh Penguins    
778 2013-06-30        Edmonton Oilers    
779 2013-06-30         Buffalo Sabres    
780 2013-06-30      Detroit Red Wings    

                                       Team_A_Acquires                Team_B  \
760  Tomas Vokoun · $2,000,000$2,000,000$2,000,000S...  Washington Capitals    
761  Chris Bourque · $0$0$0 (Signing Rights)Sum: $0...  Washington Capitals    
762                              Future Considerations   New York Islanders    
763                                   Sebastian Erixon    Vancouver Canucks    
764                       Cody HodgsonAlexander Sulzer    Vancouver Canucks    
765                                      Dale Mitchell  Toronto Maple Leafs    
766                                         Greg Zanon       Minnesota Wild    
769  Benn Ferriero · $0$550,000$550,000 (AHL/JR)201...     New York Rangers    
770  2013 2nd round pick (SJS - #50 - Dillon Heathe...  Pittsburgh Penguins    
771  2013 7th round pick (MTL - #206 - MacKenzie We...   Montreal Canadiens    
772  2013 3rd round pick (STL - #83 - Bogdan Yakimo...      St. Louis Blues    
773  2013 3rd round pick (WPG - #74 - John Hayden)2...        Winnipeg Jets    
774  2013 2nd round pick (CHI - #61 - Zachary Sanfo...        Winnipeg Jets    
775  Nino Niederreiter · $870,000$870,000$810,000Su...   New York Islanders    
776  2013 2nd round pick (NJD - #39 - Laurent Dauph...    New Jersey Devils    
777  2013 2nd round pick (SJS - #50 - Dillon Heathe...      San Jose Sharks    
778  2013 2nd round pick (LAK - #57 - William Carri...    Los Angeles Kings    
779  Jamie McBain · $875,000$1,800,000$1,900,000 (A...  Carolina Hurricanes    
780  2013 1st round pick (SJS - #20 - Anthony Manth...      San Jose Sharks    

                                       Team_B_Acquires Team_C Team_C_Acquires  
760  2012 7th round pick (PIT - #203 - Sergei Koste...    NaN             NaN  
761  Zach Hamill · $700,000$700,000$700,000Sum: $70...    NaN             NaN  
762                                    Yury Alexandrov    NaN             NaN  
763                                      Andrew Gordon    NaN             NaN  
764                    Zack KassianMarc-Andre Gragnani    NaN             NaN  
765                                        Mark Fraser    NaN             NaN  
766                                     Steven Kampfer    NaN             NaN  
769  Justin Falk · $975,000$975,000$975,000Sum: $97...    NaN             NaN  
770  2013 2nd round pick (CBJ - #44 - Tristan Jarry...    NaN             NaN  
771  2014 7th round pick (FLA - #182 - Hugo Fagerbl...    NaN             NaN  
772  2013 2nd round pick (LAK - #57 - William Carri...    NaN             NaN  
773  Michael Frolík · $2,333,333$2,333,333$1,900,00...    NaN             NaN  
774  2013 3rd round pick (WSH - #84 - James Lodge)2...    NaN             NaN  
775  Cal Clutterbuck · $2,750,000$2,750,000$2,000,0...    NaN             NaN  
776  2013 2nd round pick (ARI - #42 - Steven Santin...    NaN             NaN  
777  Tyler Kennedy · $2,350,000$2,350,000$2,350,000...    NaN             NaN  
778  2013 2nd round pick (EDM - #37 - Valentin Zyko...    NaN             NaN  
779  Andrej Sekera · $2,750,000$2,750,000$1,750,000...    NaN             NaN  
780  2013 1st round pick (DET - #18 - Mirco Muel

In [148]:
trades_13 = (raw['Date']>='2013-1-1') & (raw['Date']<='2013-4-3')
raw13 = raw.loc[trades_13]

In [149]:
trades_14 = (raw['Date']>='2014-1-1') & (raw['Date']<='2014-3-3')
raw14 = raw.loc[trades_14]

In [150]:
trades_15 = (raw2['Date']>='2015-1-1') & (raw2['Date']<='2015-3-2')
raw15 = raw2.loc[trades_15]

In [109]:
raw[raw['Date'].dt.year == 2015]

Date                  Team_A  \
980  2015-06-30          Anaheim Ducks    
981  2015-06-30        Arizona Coyotes    
982  2015-06-30     Chicago Blackhawks    
983  2015-06-30          Boston Bruins    
984  2015-06-30        San Jose Sharks    
985  2015-06-29          Boston Bruins    
986  2015-06-27        Arizona Coyotes    
987  2015-06-27          Anaheim Ducks    
988  2015-06-27        Arizona Coyotes    
989  2015-06-27     Chicago Blackhawks    
990  2015-06-27        Edmonton Oilers    
991  2015-06-27       Florida Panthers    
992  2015-06-27          Boston Bruins    
993  2015-06-27        Edmonton Oilers    
994  2015-06-27      Los Angeles Kings    
995  2015-06-27        Edmonton Oilers    
996  2015-06-27        Edmonton Oilers    
997  2015-06-27        San Jose Sharks    
998  2015-06-27       New York Rangers    
999  2015-06-27          Anaheim Ducks    
1000 2015-06-27        Edmonton Oilers    
1001 2015-06-27          Anaheim Ducks    
1002 2015-06-27           Dallas Stars    
1003 2015-06-27      New Jersey Devils    
1088 2015-12-28        Edmonton Oilers    
1089 2015-12-16        Arizona Coyotes    
1090 2015-12-14     Chicago Blackhawks    
1091 2015-11-12         Calgary Flames    
1092 2015-10-21  Columbus Blue Jackets    
1093 2015-10-06      Los Angeles Kings    
1094 2015-10-04         Calgary Flames    
1095 2015-09-17     New York Islanders    
1096 2015-09-11    Carolina Hurricanes    
1097 2015-09-09        Arizona Coyotes    
1098 2015-07-28    Pittsburgh Penguins    
1099 2015-07-23      New Jersey Devils    
1100 2015-07-12    Nashville Predators    
1101 2015-07-10     Chicago Blackhawks    
1102 2015-07-06     Chicago Blackhawks    
1103 2015-07-02        St. Louis Blues    
1104 2015-07-01    Pittsburgh Penguins    
1105 2015-07-01          Boston Bruins    
1106 2015-07-01    Nashville Predators    
1107 2015-07-01         Calgary Flames    
1108 2015-07-01     Montreal Canadiens    

                                        Team_A_Acquires  \
980   Kevin Bieksa · $4,600,000$4,600,000$2,500,000S...   
981   Boyd Gordon · $2,050,000$3,000,000$3,000,000 (...   
982   Artem Anisimov · $3,283,333$3,283,333$4,000,00...   
983   Sean Kuraly · $0$0$02016 1st round pick (SJS -...   
984   2016 7th round pick (STL - #209 - Nikolaj Krag...   
985   Zac Rinaldo · $850,000$850,000$850,000Sum: $85...   
986   2015 3rd round pick (WSH - #83 - Jens Looke)20...   
987   2015 2nd round pick (FLA - #41 - Ryan Gropp)20...   
988   Nicklas Grossmann ($500,000 retained - 14.3%) ...   
989   Ryan Haggerty · $0$925,000$70,000 (AHL/JR)Sum:...   
990   2015 7th round pick (ANA - #208 - Miroslav Svo...   
991   2016 5th round pick (NYI - #140 - Cole Candell...   
992   2016 5th round pick (MIN - #136 - Cameron Clar...   
993   Eric Gryba · $1,250,000$1,250,000$1,300,000Sum...   
994   2015 4th round pick (CBJ - #99 - Austin Wagner...   
995   Brad Ross · $0$0$0 (Signing Rights)2015 4th ro...   
996   Todd McLellan (Coach) (Compensation from the O...   
997   Patrick McNally · $0$0$0Sum: $0$0$0Change: $0C...   
998   2015 4th round pick (WSH - #113 - Brad Morriso...   
999   Anton Khudobin · $2,250,000$2,250,000$2,500,00...   
1000  Cam Talbot · $1,450,000$1,450,000$1,450,000201...   
1001  Carl Hagelin · $4,000,000$4,000,000$5,000,0002...   
1002  Antti Niemi · $0$0$4,500,000Sum: $0$0$4,500,00...   
1003  2015 2nd round pick (DAL - #42 - Mackenzie Bla...   
1088  Zack Kassian · $1,750,000$1,750,000$2,000,000S...   
1089  Christian Thomas · $0$600,000$105,000 (AHL/JR)...   
1090  Rob Scuderi ($1,113,750 retained - 33%) · $2,2...   
1091  Kevin Poulin · $0$650,000$275,000 (AHL/JR)Sum:...   
1092  John Tortorella (Coach)Sum: $0$0$0Change: $0Ch...   
1093  2017 7th round pick (NJD - #191 - Jocktan Chai...   
1094  Freddie Hamilton · $687,500$687,500$687,500Sum...   
1095  Matthew Finn · $0$827,500$70,000 (AHL/JR)Chris...   
1096  Joakim Nordström · $0$605,000$605,000 (AHL/JR)...   
1097  Stefan Elliott · $0$

In [151]:
trades_16 = (raw['Date']>='2016-1-1') & (raw['Date']<='2016-2-29')
raw16 = raw.loc[trades_16]

In [152]:
trades_17 = (raw['Date']>='2017-1-1') & (raw['Date']<='2017-3-1')
raw17 = raw.loc[trades_17]

In [153]:
trades_18 = (raw2['Date']>='2018-1-1') & (raw2['Date']<='2018-2-26')
raw18 = raw2.loc[trades_18]

In [114]:
raw[raw['Date'].dt.year == 2018]

Date                  Team_A  \
1221 2018-06-30     Montreal Canadiens    
1222 2018-06-30          Anaheim Ducks    
1223 2018-06-27         Buffalo Sabres    
1224 2018-06-27     Chicago Blackhawks    
1225 2018-06-24        Arizona Coyotes    
1226 2018-06-24     Chicago Blackhawks    
1227 2018-06-23        Arizona Coyotes    
1228 2018-06-23        Arizona Coyotes    
1229 2018-06-23     Colorado Avalanche    
1230 2018-06-23         Buffalo Sabres    
1231 2018-06-23        Edmonton Oilers    
1232 2018-06-23       Florida Panthers    
1233 2018-06-23     Chicago Blackhawks    
1234 2018-06-23    Carolina Hurricanes    
1235 2018-06-23    Pittsburgh Penguins    
1236 2018-06-23     Montreal Canadiens    
1237 2018-06-23      Vancouver Canucks    
1238 2018-06-23  Columbus Blue Jackets    
1239 2018-06-23     Montreal Canadiens    
1240 2018-06-23         Calgary Flames    
1241 2018-06-23        Edmonton Oilers    
1242 2018-06-23         Calgary Flames    
1243 2018-06-22        St. Louis Blues    
1244 2018-06-22       New York Rangers    
1355 2018-12-30     Chicago Blackhawks    
1356 2018-12-30        Edmonton Oilers    
1357 2018-12-29       Florida Panthers    
1358 2018-12-28          Anaheim Ducks    
1359 2018-12-10          Anaheim Ducks    
1360 2018-12-05        Ottawa Senators    
1361 2018-12-03          Anaheim Ducks    
1362 2018-12-03    Toronto Maple Leafs    
1363 2018-11-27         Calgary Flames    
1364 2018-11-25        Arizona Coyotes    
1365 2018-11-22        Edmonton Oilers    
1366 2018-11-21      Los Angeles Kings    
1367 2018-11-16        Edmonton Oilers    
1368 2018-11-14      Los Angeles Kings    
1369 2018-11-10         Buffalo Sabres    
1370 2018-10-18          Anaheim Ducks    
1371 2018-10-03         Minnesota Wild    
1372 2018-10-01        Edmonton Oilers    
1373 2018-10-01           Dallas Stars    
1374 2018-10-01         Buffalo Sabres    
1375 2018-10-01         Calgary Flames    
1376 2018-09-13        Ottawa Senators    
1377 2018-09-11          Boston Bruins    
1378 2018-09-10     Montreal Canadiens    
1379 2018-08-20         Calgary Flames    
1380 2018-08-02         Buffalo Sabres    
1381 2018-07-18        Arizona Coyotes    
1382 2018-07-12        Arizona Coyotes    
1383 2018-07-03     New York Islanders    
1384 2018-07-01         Buffalo Sabres    

                                        Team_A_Acquires  \
1221  Joel Armia · $1,850,000$1,850,000$1,850,000Ste...   
1222  Chase De Leo · $650,000$650,000$650,000Sum: $6...   
1223  Matt Hunwick · $2,250,000$2,250,000$2,500,000C...   
1224  Jordan Schroeder · $0$650,000$650,000 (AHL/JR)...   
1225  Robbie Russo · $0$650,000$650,000 (AHL/JR)Sum:...   
1226  Michael Chaput · $0$675,000$275,000 (AHL/JR)Su...   
1227  2018 4th round pick (SJS - #114 - Ivan Prosvet...   
1228  2018 3rd round pick (TOR - #87 - Linus Karlsso...   
1229  2018 3rd round pick (OTT - #64 - Justus Annune...   
1230  2019 6th round pick (TOR - #177 - Gustav Bergl...   
1231  2018 2nd round pick (WSH - #62 - Olivier Rodri...   
1232  2018 3rd round pick (NSH - #89 - Logan Hutsko)...   
1233  2018 5th round pick (FLA - #139 - Mikael Hakka...   
1234  2019 7th round pick (BOS - #216 - Massimo Rizz...   
1235  2019 7th round pick (VGK - #203 - Valtteri Puu...   
1236  2018 7th round pick (MTL - #190 - Brett Staple...   
1237  2018 6th round pick (WSH - #186 - Artyom Manuk...   
1238  2018 6th round pick (MTL - #159 - Tim Berni)Su...   
1239  2018 5th round pick (FLA - #139 - Mikael Hakka...   
1240  2018 4th round pick (WPG - #122 - Milos Roman)...   
1241  Hayden Hawkey · $0$0$0Sum: $0$0$0Change: $0Cha...   
1242  Elias Lindholm · $4,850,000$4,850,000$4,850,00...   
1243  2018 1st round pick (TOR - #25 - Dominik Bokk)...   
1244  2018 1st round pick (PIT - #22 - K'Andre Mille...   
1355  Jason Garrison · $650,000$650,000$650,000Drake...   
1356  Alex Petrovic · $1,950,000$1,950,000$1,950,000...   
1357  2020 5th round pick (TOR - #137 - Dmitry Ovchi

In [154]:
trades_19 = (raw['Date']>='2019-1-1') & (raw['Date']<='2019-2-25')
raw19 = raw.loc[trades_19]

In [155]:
trades_20 = (raw['Date']>='2020-1-1') & (raw['Date']<='2020-2-24')
raw20 = raw.loc[trades_20]

In [156]:
trades_21 = (raw['Date']>='2021-1-1') & (raw['Date']<='2021-4-12')
raw21 = raw.loc[trades_21]

In [157]:
trades_22 = (raw['Date']>='2022-1-1') & (raw['Date']<='2022-3-21')
raw22 = raw.loc[trades_22]

In [158]:
trades_23 = (raw['Date']>='2023-1-1') & (raw['Date']<='2023-3-3')
raw23 = raw.loc[trades_23]

In [159]:
trades_24 = (raw2['Date']>='2024-1-1') & (raw['Date']<='2023-3-8')
raw24 = raw2.loc[trades_24]

In [184]:
new_raw = pd.concat([raw06,raw07,raw08,raw09,raw10,raw11,raw12,raw13,raw14,raw15,raw16,raw17,raw18,raw19,raw20,raw21,raw22,raw23,raw24],axis=0)
new_raw = new_raw.reset_index()
new_raw

index       Date               Team_A  \
0       32 2006-03-09     Arizona Coyotes    
1       33 2006-03-09       Anaheim Ducks    
2       34 2006-03-09    Florida Panthers    
3       35 2006-03-09     Arizona Coyotes    
4       36 2006-03-09  Chicago Blackhawks    
..     ...        ...                  ...   
913    320 2024-01-31      Calgary Flames    
914    321 2024-01-25      Minnesota Wild    
915    322 2024-01-11      Buffalo Sabres    
916    323 2024-01-08       Anaheim Ducks    
917    324 2024-01-06  Chicago Blackhawks    

                                       Team_A_Acquires                Team_B  \
0                                        Yanick Lehoux    Los Angeles Kings    
1                                       Sean O'Donnell      Arizona Coyotes    
2                                      Richard Jackman  Pittsburgh Penguins    
3                                         Jamie Rivers    Detroit Red Wings    
4       2006 6th round pick (MTL - #169 - Chris Auger)   Montreal Canadiens    
..                                                 ...                   ...   
913  Andrei Kuzmenko · $5,500,000$5,500,000$5,500,0...    Vancouver Canucks    
914  Will Butcher · $0$775,000$425,000 (AHL/JR)Sum:...  Pittsburgh Penguins    
915  Future ConsiderationsSum: $0$0$0Change: $0Chan...   Montreal Canadiens    
916  Cutter Gauthier · $0$0$0 (AHL/JR) (Signing Rig...  Philadelphia Flyers    
917  Rem Pitlick · $0$1,100,000$1,100,000 (AHL/JR)S...  Pittsburgh Penguins    

                                       Team_B_Acquires Team_C Team_C_Acquires  
0                                          Tim Jackman    NaN             NaN  
1                                        Joel Perrault    NaN             NaN  
2                                         Petr Taticek    NaN             NaN  
3       2006 7th round pick (ARI - #191 - Nick Oslund)    NaN             NaN  
4                                         Todd Simpson    NaN             NaN  
..                                                 ...    ...             ...  
913  Elias Lindholm · $4,850,000$4,850,000$4,850,00...    NaN             NaN  
914  Maxim Cajkovic · $0$850,833$70,000 (AHL/JR)Sum...    NaN             NaN  
915  Filip Cederqvist · $0$850,000$70,000 (AHL/JR)S...    NaN             NaN  
916  Jamie Drysdale · $2,300,000$2,300,000$2,300,00...    NaN             NaN  
917  2026 7th round pick (CHI) [Conditional]Sum: $0...    NaN             NaN  

[918 rows x 8 columns]

In [186]:
new_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 918 entries, 0 to 917
Data columns (total 8 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   index            918 non-null    int64         
 1   Date             918 non-null    datetime64[ns]
 2   Team_A           918 non-null    object        
 3   Team_A_Acquires  918 non-null    object        
 4   Team_B           918 non-null    object        
 5   Team_B_Acquires  918 non-null    object        
 6   Team_C           17 non-null     object        
 7   Team_C_Acquires  17 non-null     object        
dtypes: datetime64[ns](1), int64(1), object(6)
memory usage: 57.5+ KB


In [187]:
new_raw.to_csv('/Users/phil/Documents/Capstone/Data/deadline_trades.csv')

After pulling the correct dates for each season - which I've defined as January 1st to each season's trade deadline (about 75% through the season), we have 900+ trades between 2006 and 2024 to work with in the proper format.  We have saved this as a new csv to work in another notebook.